doc/developer_guide/index.rst

```diff
@@ -85,25 +86,57 @@ full test suite requires a wide range of dependencies. To make it
 easier to install and run different parts of the test suite across
 different platforms hvPlot uses a library called ``pyctdev`` to make things
 more consistent and general. Specify also the desired Python version you want
-to base your environment on. It is advised to choose the minimum version
-currently supported by hvPlot on the main development branch.
+to base your environment on.
+
+You will need to pick a Python version. The best practice is to choose the minimum version
+currently supported by hvPlot on the main development branch. If you cannot get the minimum
+version installed, then try with a more recent version of Python.
 
 .. code-block:: sh
 
-    conda install python=3.x pyctdev
+    mamba install python=3.x pyctdev
 
 Finally to install the dependencies required to run the full unit test
 suite and all the examples:
 
 .. code-block:: sh
 
-    doit develop_install -o tests -o examples
+    doit develop_install -o tests -o examples --conda-mode mamba
 
 Add ``-o doc`` if you want to install the dependencies required to build
 the website.
 
 .. _devguide_python_setup:
 
+Commands
+--------
+
+You can list the available `doit` commands with `doit list`.
+
+.. code-block:: sh
+
+    $ doit list
+    build_docs             build docs
+    develop_install        python develop install, with specified optional groups of dependencies (installed by conda only).
+    ecosystem_setup        Common conda setup (must be run in base env).
+    env_capture            Report all information required to recreate current conda environment
+    env_create             Create named environment if it doesn't already exist
+    env_dependency_graph   Write out dependency graph of named environment.
+    env_export             Generate a pinned environment.yaml from specified env, filtering
+    env_export2            
+    list_envs              
+    miniconda_download     Download Miniconda3-latest
+    miniconda_install      Install Miniconda3-latest to location if not already present
+    package_build          Build and then test conda.recipe/ (or specified alternative).
+    package_test           Test existing package
+    package_upload         Upload package built from conda.recipe/ (or specified alternative).
+    pip_on_conda           Experimental: provide pip build env via conda
+    test_all               Run all tests
+    test_examples          Test that default examples run
+    test_flakes            Flake check python and notebooks
+    test_unit              Run unit tests with coverage
```

Comment:
Note to myself, this is not very useful as contributors won't use most of these commands. Documenting `doit` could be done on the HoloViz site, not even sure we want to do that really.

Reply:
I put it in because currently most of the info on how to build, serve or test stuff is not documented. So this is the best there is to my knowledge.

It might be obvious to you how to do this. But an outsider like me, not having access to a colleague with the knowledge etc have no chance of figuring out how to build or view the docs currently for example.

Reply:
There will be soon a contributing guide that should make more of this knowledge available to everyone: https://github.com/holoviz/holoviz/pull/323

In the mean time, the source of truth for running the tests/building the docs/building the packages are the **Github Action Workflows**.